### PR TITLE
Westmidlands Class 3 Mimi K - Github - Week 1

### PR DESCRIPTION
**Pull request by mimi2111**
# Your Details

- Your Name: Mimi K
- Your City: West Midlands, Birmingham
- Your Slack Name: Mimi

# Homework Details

- Module: Github
- Week: 1

-----
[View rendered homework.md](https://github.com/mimi2111/GitHomeworkFixErrors/blob/main/homework.md)

### CODE REVIEW

File: homework.md [hunk 1]
```diff
@@ -4,77 +4,78 @@
...
 What is JavaScript?
 
-A programming language used on the internet
+A programming language for the web
 
 ## Question 8
 
 What three problem does Git & Github solve?
 
-When people want to show off code to each other they can put it on Github
```
Comment (Pezhman-Azizi) [inline homework.md:53]: great response ;)

File: homework.md [hunk 2]
```diff
@@ -4,77 +4,78 @@
...
 What happens when you `fork` a repository?
 
-You delete it
+it take a repository from someone account to my account
 
-## Question 10 
+## Question 10
 
-What happens when you clone a repostory?
+What happens when you clone a repository?
```
Comment (Pezhman-Azizi) [inline homework.md:64]: wow I have not seen that lol

File: homework.md [hunk 3]
```diff
@@ -4,77 +4,78 @@
...
 
-It send it to a friend
+It to produce a copies of something
```
Comment (Pezhman-Azizi) [inline homework.md:66]: that is correct and it also creates a whole copy of the original version down to your local computer.

File: homework.md [hunk 4]
```diff
@@ -4,77 +4,78 @@
...
 
 ## Question 11
 
 What is a Pull Request?
 
-When you send a file over the internet
+mean to ask for change committed to their own record to be considered
```
Comment (Pezhman-Azizi) [inline homework.md:72]: I think you ask your project to be reviewed. Please correct me if I am wrong.

File: homework.md [hunk 5]
```diff
@@ -4,77 +4,78 @@
...
+h1 {
 }
 ```
 
 ## Question 7
 
 What is JavaScript?
 
-A programming language used on the internet
+A programming language for the web
```
Comment (MireilleNyoubissi) [inline homework.md:47]: One particular aspect of javascript, if not the most important one, is that it makes web pages interactive. I think is incomplete to define JavaScript without mentioning that it allows us to program the behaviour of the web pages.

File: homework.md [hunk 6]
```diff
@@ -4,77 +4,78 @@
...
 
 ## Question 8
 
 What three problem does Git & Github solve?
 
-When people want to show off code to each other they can put it on Github
+1 they make collaboration very easy with others doc
+2 help change some code
+3 track information and help you to save your history
```
Comment (MireilleNyoubissi) [inline homework.md:55]: also, version control

File: homework.md [hunk 7]
```diff
@@ -4,77 +4,78 @@
...
 
 ## Question 9
 
 What happens when you `fork` a repository?
 
-You delete it
+it take a repository from someone account to my account
```
Comment (MireilleNyoubissi) [inline homework.md:61]: To be more explicit, I would say to fork a repository is to make your own copy of that repository from one account to yours.  It is important to emphasize that the point here is to find a way to freely implement your changes without affecting the original repository.

File: homework.md (hunk 3 above)
Comment (MireilleNyoubissi) [reply homework.md:66]: Yes, and I would also add that your local copy will retain a link to its original repository and, if you attempt to implement your changes, you may face protection measures.

File: homework.md (hunk 4 above)
Comment (MireilleNyoubissi) [reply homework.md:72]: I would say it is a method of submitting contributions to an open development project.

File: homework.md [hunk 8]
```diff
@@ -86,28 +87,28 @@ You send the changes of your local repository to Github
 
 What is Github?
 
-A place for us to store and share our code.
+is a source control platform use when created code .is a store place where you can save your project
```
Comment (MireilleNyoubissi) [inline homework.md:91]: I am not sure whether you need Github to be able to create code. Anyway, You should look for a better definition. 
I can suggest this definition: " it is a Git repository hosting service, that also provides access control and several collaboration features". But, there is always a better one so, please do your own research.

File: homework.md [hunk 9]
```diff
@@ -86,28 +87,28 @@ You send the changes of your local repository to Github
...
 
 ## Question 15
 
 What does it mean to "Commit your changes"?
 
-Send you code to Github
+is to solve the error your change to the repository or stash your change later
```
Comment (MireilleNyoubissi) [inline homework.md:97]: to put it simply, "to commit your changes" is a Github way of saying to "save your changes". And, don't forget that it is good practice to always give a name to your changes and a brief description before you commit.

File: homework.md [hunk 10]
```diff
@@ -86,28 +87,28 @@ You send the changes of your local repository to Github
...
 
-Microsoft Word
+Visual studio Code
 
 ## Question 17
 
 What do you use "Commit Messages" for?
 
-To record what changes you made to the code
+when things really tend to go south the revision history is a great resource to find out quickly where exactly things went wrong
```
Comment (MireilleNyoubissi) [inline homework.md:109]: I believe we use commit messages to identify our changes. They provide information about what has changed and why.

File: homework.md [hunk 11]
```diff
@@ -86,28 +87,28 @@ You send the changes of your local repository to Github
...
 
 ## Question 18
 
 Where can I look to find help with Git?
 
-The Git Cheatsheet!
+The Git help
```
Comment (MireilleNyoubissi) [inline homework.md:114]: also, Git Documentation.

**Review by MireilleNyoubissi — COMMENTED** [review]
Good answers Mimi, I hope my contributions would enhance your Git/Github knowledge. It was a pleasure reviewing your work. Keep the hard work up.

**Comment by Kev-One** [comment]
well done Mimi. great answers

File: homework.md (hunk 7 above)
Comment (Kev-One) [reply homework.md:61]: awesome! well done Mireille

**Pull request closed without merging**